src/tools/ScenarioMeasurement/Startup/TimeToMain2Parser.cs

```diff
@@ -1,26 +1,29 @@
-﻿using Microsoft.Diagnostics.Tracing;
-using Microsoft.Diagnostics.Tracing.Parsers;
-using Reporting;
-using System;
+﻿using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
+using Microsoft.Diagnostics.Tracing;
+using Reporting;
+using ScenarioMeasurement.TraceEventParsers;
 
 namespace ScenarioMeasurement;
 
 public class TimeToMain2Parser : IParser
 {
-    private const string EventSourceName = PerfLabValues.EventSourceName;
-    private const string OnMainEventName = PerfLabValues.OnMainEventName;
-    private const int OnMainEventId = PerfLabValues.OnMainEventId;
-
     public void EnableKernelProvider(ITraceSession kernel)
     {
         kernel.EnableKernelProvider(TraceSessionManager.KernelKeyword.Process, TraceSessionManager.KernelKeyword.Thread, TraceSessionManager.KernelKeyword.ContextSwitch);
     }
 
     public void EnableUserProviders(ITraceSession user)
     {
-        user.EnableUserProvider(EventSourceName, TraceEventLevel.Verbose);
+        user.EnableUserProvider(PerfLabValues.EventSourceName, TraceEventLevel.Verbose);
+        if (user is LinuxTraceSession)
+        {
+            user.AddRawEvent($"{PerfLabValues.LTTngProviderName}:{PerfLabValues.OnMainEventName}");
+        }
+        var baseDir = Environment.GetEnvironmentVariable("HELIX_CORRELATION_PAYLOAD") ?? Path.GetFullPath("..");
+        Startup.AddTestProcessEnvironmentVariable("DOTNET_STARTUP_HOOKS", Path.Join(baseDir, PerfLabValues.ForwarderName, $"{PerfLabValues.ForwarderName}.dll"));
```

Comment:
I think `DOTNET_STARTUP_HOOKS` should only be set on Linux.

Reply:
As we discussed offline, I want to keep the Linux and Windows versions close to each other, so even on Windows the startup hook is used, but only on Linux it is emitting the LTTng events ([here](https://github.com/dotnet/performance/pull/3170/files#diff-87862d1da7576ce7a9dbc519aadcdfad0ec8626cc81e66c30c5bb5ada317441aR20)).

Reply:
Yup, makes total sense to me.